### PR TITLE
support ref for item of double array

### PR DESCRIPTION
**Pull request by duguhaotian**
test data
```
[
    [
        {"item1": "test11", "item2": 11, "item3": true},
        {"item1": "test12", "item2": 12, "item3": false}
    ],
    [
        {"item1": "test21", "item2": 21, "item3": false},
        {"item1": "test22", "item2": 22, "item3": true}
    ]
]
```

json schema
```
$ cat tests/test-spec/basic/test_top_double_array_refobj.json 
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "top double array of ref object",
  "type": "array",
  "items": {
      "type": "array",
      "items": {
        "$ref": "test_double_array_item.json"
      }
  }
}
````

generate code

```
$ cat src/basic_test_top_double_array_refobj.h
// Generated from test_top_double_array_refobj.json. Do not edit!
#ifndef BASIC_TEST_TOP_DOUBLE_ARRAY_REFOBJ_SCHEMA_H
#define BASIC_TEST_TOP_DOUBLE_ARRAY_REFOBJ_SCHEMA_H

#include <sys/types.h>
#include <stdint.h>
#include "json_common.h"
#include "basic_test_double_array_item.h"

#ifdef __cplusplus
extern "C" {
#endif

typedef struct {
    basic_test_double_array_item ***items;
    size_t *subitem_lens;

    size_t len;

}
basic_test_top_double_array_refobj_container;

void free_basic_test_top_double_array_refobj_container (basic_test_top_double_array_refobj_container *ptr);

basic_test_top_double_array_refobj_container *basic_test_top_double_array_refobj_container_parse_file(const char *filename, const struct parser_context *ctx, parser_error *err);

basic_test_top_double_array_refobj_container *basic_test_top_double_array_refobj_container_parse_file_stream(FILE *stream, const struct parser_context *ctx, parser_error *err);

basic_test_top_double_array_refobj_container *basic_test_top_double_array_refobj_container_parse_data(const char *jsondata, const struct parser_context *ctx, parser_error *err);

char *basic_test_top_double_array_refobj_container_generate_json(const basic_test_top_double_array_refobj_container *ptr, const struct parser_context *ctx, parser_error *err);

#ifdef __cplusplus
}
#endif

#endif
```

Signed-off-by: duguhaotian <knowledgehao@163.com>

### CODE REVIEW

**Comment by duguhaotian** [comment]
@giuseppe hi, pls check this pr

File: src/headers.py [hunk 1]
```diff
@@ -178,7 +178,11 @@ def header_reflect_top_array(obj, prefix, header):
     c_typ = helpers.get_prefixed_pointer(obj.name, obj.subtyp, prefix) or \
         helpers.get_map_c_types(obj.subtyp)
     if obj.subtypobj is not None:
-        c_typ = helpers.get_name_substr(obj.name, prefix) + " *"
+        if obj.doublearray and obj.subtypname is not None:
+            print(obj.__dict__)
```
Comment (giuseppe) [inline src/headers.py:182]: do we need this print or was it used for debugging?

Comment (duguhaotian) [reply src/headers.py:182]: sorry, i forget....

Comment (duguhaotian) [reply src/headers.py]: fixed

**Review by giuseppe — APPROVED** [review]
LGTM

**Pull request merged**